securedrop/manage.py

```diff
@@ -314,6 +316,60 @@ def translate(args):
         """.format(translations_dir=args.translations_dir))
 
 
+def translate_desktop(args):
+    messages_file = os.path.join(args.translations_dir, 'desktop.pot')
+
+    if args.extract_update:
+        sh("""
+        set -xe
+        cd {translations_dir}
+        xgettext \
+           --output=desktop.pot \
+           --language=Desktop \
+           --keyword \
+           --keyword=Name \
+           --package-version={version} \
+           --msgid-bugs-address='securedrop@freedom.press' \
+           --copyright-holder='Freedom of the Press Foundation' \
+           {sources}
+
+        # remove this line so the file does not change if no
+        # strings are modified
+        sed -i '/^"POT-Creation-Date/d' {messages_file}
+        """.format(translations_dir=args.translations_dir,
+                   messages_file=messages_file,
+                   version=args.version,
+                   sources=" ".join(args.source)))
+
+        changed = subprocess.call("git diff --quiet {}".format(messages_file),
+                                  shell=True)
+
+        if changed:
+            for f in os.listdir(args.translations_dir):
+                if not f.endswith('.po'):
+                    continue
+                po_file = os.path.join(args.translations_dir, f)
+                sh("""
+                msgmerge --update {po_file} {messages_file}
+                """.format(po_file=po_file,
+                           messages_file=messages_file))
+            log.warning("messages translations updated in " + messages_file)
+        else:
+            log.warning("desktop translations are already up to date")
+
+    if args.compile:
+        sh("""
+        set -ex
+        cd {translations_dir}
+        ls *.po | sed -e 's/\.po$//' > LINGUAS
```

Comment:
Agreed, should be `find -type f -iname '*.po' | sed -e 's/\.po$//' > LINGUAS` or similar.

Reply:
Because I was curious if this actually was in `shellcheck`:
```
heartsucker@pythagoras:/tmp$ echo -e '#!/bin/bash\nls | sed wat' > test.sh
heartsucker@pythagoras:/tmp$ shellcheck test.sh
In test.sh line 2:
ls | sed wat
^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```